packages/Kokkos/src/DTK_PointCloud_decl.hpp

```diff
@@ -0,0 +1,116 @@
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \brief DTK_PointCloud_decl.hpp
+ * \author Stuart R. Slattery
+ * \brief Point cloud.
+ */
+//---------------------------------------------------------------------------//
+
+#ifndef DTK_POINTCLOUD_DECL_HPP
+#define DTK_POINTCLOUD_DECL_HPP
+
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_Comm.hpp>
+
+#include <Kokkos_View.hpp>
+
+namespace DataTransferKit
+{
+//---------------------------------------------------------------------------//
+/*!
+  \class PointCloud
+  \brief Point cloud.
+*/
+//---------------------------------------------------------------------------//
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+class PointCloud
+{
+  public:
+
+    //@{
+    //! Type aliases.
+    using scalar_type = Scalar;
+    using local_ordinal_type = LocalOrdinal;
+    using global_ordinal_type = GlobalOrdinal;
+    using node_type = Node;
+    using device_type = typename Node::device_type;
+    using execution_space = typename DeviceType::execution_space;
+    using GlobalIdView = Kokkos::View<global_ordinal_type*,device_type>;
+    using CoordinateView = Kokkos::View<scalar_type**,device_type>;
+    //@}
+
+    // Constructor.
+    PointCloud( const Teuchos::RCP<const Teuchos::Comm<int> >& comm,
+                const GlobalIdView global_ids,
+                const CoordinateView coordinates );
+
+    // Get the spatial dimension of the point cloud.
+    int spatialDimension() const;
+
+    // Get the local number of points in the point cloud.
+    LocalOrdinal localNumberOfPoints() const;
+
+    // Get the global number of points in the point cloud.
+    GlobalOrdinal globalNumberOfPoints() const;
+
+    // Get the point global ids.
+    const GlobalIdView globalIds() const;
+
+    // Get the point coordinates.
+    const CoordinateView coordinates() const;
+
+  private:
+
+    // Communicator.
+    Teuchos::RCP<const Teuchos::Comm<int> > _comm;
```

Comment:
no - I want to stick with _comm per our numerous previous discussions on this issue. I find it much more clear that this is class data and it is an extremely common convention. I would rather go back to d_comm than comm_.